### PR TITLE
Dynamic joint handles

### PR DESCRIPTION
**Pull request by jordan-palacios**
The intention of this PR is to add a new type of joint handle and the interfaces to handle them. They are intended to replace `JointStateHandle` and `JointCommandHandle` eventually. For the duration of this PR, however, they will coexist.

See #124 for more information.

### CODE REVIEW

**Comment by destogl** [comment]
this is a good transition PR for the Joint/Sensor components. I agree to go on with this.

File: hardware_interface/test/test_robot_hardware_interface.cpp [hunk 1]
```diff
@@ -120,12 +120,12 @@ TEST_F(TestRobotHardwareInterface, cannot_double_register_handles)
 TEST_F(TestRobotHardwareInterface, can_get_registered_joint_names)
 {
   SetUpHandles();
-  EXPECT_THAT(robot_.get_registered_joint_names(), ElementsAre(JOINT_NAME));
+  // EXPECT_THAT(robot_.get_registered_joint_names(), ElementsAre(JOINT_NAME));
```
Comment (Karsten1987) [inline hardware_interface/test/test_robot_hardware_interface.cpp:123]: What's the status of these tests?

**Comment by Karsten1987** [comment]
With #121 being merged, is this going to be reworked to use the same `Joint` and `Sensor` interfaces?

File: hardware_interface/test/test_robot_hardware_interface.cpp (hunk 1 above)
Comment (jordan-palacios) [reply hardware_interface/test/test_robot_hardware_interface.cpp:123]: They only test the `OperationModeHandle`. The joint handles tests are now in `test_joint_handle.cpp` and `test_register_joints.cpp`.

**Review by bmagyar — APPROVED** [review]

**Pull request merged**